Source/WebKit2/NetworkProcess/soup/NetworkProcessSoup.cpp

```diff
@@ -105,10 +105,7 @@ void NetworkProcess::userPreferredLanguagesChanged(const Vector<String>& languag
 
 void NetworkProcess::setProxies(WebCore::SessionID sessionID, const Vector<WebCore::Proxy>& proxies)
 {
-    if (auto *storageSession = NetworkStorageSession::storageSession(sessionID))
-        storageSession->soupNetworkSession().setProxies(proxies);
-     else
-        SoupNetworkSession::defaultSession().setProxies(proxies);
+    NetworkStorageSession::defaultStorageSession().getOrCreateSoupNetworkSession().setProxies(proxies);
```

Comment:
This is still not correct. Now you are setting the proxies always for the default session, so private browsing sessions will not use the proxy. You could get the right session for the given sessionID, but it would still be wrong or at least not enough. I don't think this message should receive any session ID, I assume you don't really want to use different proxy settings per session. Proxy settings are global, which means we want to apply them to all existing sessions, but also to the newly created ones. So we need to store the settings globally, then apply them to all sessions (there's NetworkStorageSession::forEach for that) and also apply them when new sessions are created. I think you should do this in the web process too, since unfortunately there are still a few things loaded there. All this is done in patch attached to bug https://bugs.webkit.org/show_bug.cgi?id=128674 that I plan to land soon upstream.